### PR TITLE
Hide pytest from code snippets

### PR DESCRIPTION
**Pull request by crusaderky**
- xref #8146

Clean up code snippets when pytest starts a cluster, e.g. coiled/benchmarks:

image

### CODE REVIEW

File: distributed/distributed.yaml [hunk 1]
```diff
@@ -274,6 +274,7 @@ distributed:
       max-history: 100
       nframes: 0
       ignore-modules:
+        - _pytest
```
Comment (jrbourbeau) [inline distributed/distributed.yaml:277]: Just for my own understanding, why `_pytest` and not `pytest`?

**Review by jrbourbeau — COMMENTED** [review]
Thanks @crusaderky. This seems reasonable to me, though the `distributed/tests/test_client.py::test_computation_object_code_dask_compute` failure looks legitimate

File: distributed/distributed.yaml (hunk 1 above)
Comment (graingert) [reply distributed/distributed.yaml:277]: https://github.com/pytest-dev/pytest/blob/main/src/pytest/__init__.py doesn't have any function defs and so won't appear in any frames

Comment (graingert) [reply distributed/distributed.yaml:277]: all the function defs are in https://github.com/pytest-dev/pytest/tree/main/src/_pytest

Comment (jrbourbeau) [reply distributed/distributed.yaml:277]: TIL 🙂

File: distributed/distributed.yaml [hunk 2]
```diff
@@ -281,10 +281,17 @@ distributed:
         - coiled
         - cudf
         - cuml
+        - pluggy  # part of pytest
         - prefect
         - rechunker
         - xarray
         - xgboost
+      ignore-files:
+        - runpy\.py  # `python -m pytest` shell command
```
Comment (crusaderky) [inline distributed/distributed.yaml:290]: Known issue: this could also catch `python -m <user module>`

**Comment by crusaderky** [comment]
It turns out it was quite a bit more involved than the original one-liner. It's ready for review again.

File: distributed/distributed.yaml [hunk 3]
```diff
@@ -281,10 +281,18 @@ distributed:
...
         - rechunker
         - xarray
         - xgboost
+      ignore-files:
+        - runpy\.py  # `python -m pytest` (or other module) shell command
+        - pytest  # `pytest` shell command
+        - py\.test  # `py.test` shell command
+        - pytest-script\.py  # `pytest` shell command in Windows
+        - _pytest  # pytest implementation
+        - pycharm  # Run pytest from PyCharm GUI
```
Comment (crusaderky) [inline distributed/distributed.yaml:295]: Could someone with vscode please run test_client.py and see if there are any regressions in the code tests?

**Comment by fjetter** [comment]
I verified with vscode and added the relevant ignores.

I also incorporated https://github.com/dask/distributed/pull/8204

**Pull request merged**